src/main/java/apoc/metrics/Metrics.java

```diff
@@ -162,10 +166,18 @@ public boolean test(LoadCsv.CSVResult o) {
                     "https://neo4j.com/docs/operations-manual/current/monitoring/metrics/expose/#metrics-csv");
         }
 
-        String url = new File(metricsDir, metricName + ".csv").toURI().toString();
-        return new LoadCsv().csv(url, config)
-                .filter(Metrics.duplicatedHeaderRows)
-                .map(csvResult -> new GenericMetric(metricName, Util.toLong(csvResult.map.get("t")), csvResult.map));
+        String url = new File(metricsDir, metricName + ".csv").getAbsolutePath();
+        CountingReader reader = null;
+        try {
+            reader = FileUtils.readFile(url);
+            return new LoadCsv()
+                    .streamCsv(url, new LoadCsvConfig(config), reader)
+                    .filter(Metrics.duplicatedHeaderRows)
+                    .map(csvResult -> new GenericMetric(metricName, Util.toLong(csvResult.map.get("t")), csvResult.map));
+        } catch (Exception e) {
+            closeReaderSafely(reader);
```

Comment:
does reader get closed if an exception was not thrown?   Seems like this would be better in finally?

Reply:
good question, it's only here because of the `streamCsv` method closes the reader as the stream closes:
https://github.com/neo4j-contrib/neo4j-apoc-procedures/blob/62cb555123757e4384280246233ca0ce5539feb5/src/main/java/apoc/load/LoadCsv.java#L70

So I need to manage it only in case of the exception.
But we can change it if you want